docs/protocol/off-chain/authentication.mdx

```diff
@@ -0,0 +1,374 @@
+
+### When to Use
+
+Immediately after receiving `auth_challenge` response. This is the final step in authentication.
+
+### Prerequisites
+
+- Completed `auth_request` and received `auth_challenge`
+- Have the challenge_message
+- Have the <Tooltip content="A temporary cryptographic key delegated by a user's main wallet that provides a flexible way for the user to manage security of their funds by giving specific permissions and allowances for specific apps">session key</Tooltip> private key (client-side only)
+
+### Request
+
+| Parameter | Type | Required | Description | Example | Notes |
+|-----------|------|----------|-------------|---------|-------|
+| `challenge` | string | Yes | The challenge_message received from auth_challenge | `"550e8400-e29b-41d4-a716-446655440000"` | Must be the exact challenge from Step 2 |
+| `session_key` | string (wallet address) | No (can be derived from signature) | Session key wallet address (same as in auth_request) | `"0x9876543210fedcba9876543210fedcba98765432"` | Including it explicitly can speed up verification |
+
+### Response
+
+| Parameter | Type | Description | Example | Notes |
+|-----------|------|-------------|---------|-------|
+| `address` | string (wallet address) | Authenticated user's main wallet address | `"0x742d35Cc6634C0532925a3b844Bc9e7595f0bEb"` | Confirms which account is authenticated |
+| `session_key` | string (wallet address) | Confirmed session key wallet address | `"0x9876543210fedcba9876543210fedcba98765432"` | The authorized session key |
+| `expire` | number | Session expiration timestamp (Unix seconds) | `1699999999` | After this time, must re-authenticate |
+
+### Signature
+
+Request **MUST** be signed by the <Tooltip content="A temporary cryptographic key delegated by a user's main wallet that provides a flexible way for the user to manage security of their funds by giving specific permissions and allowances for specific apps">session key</Tooltip> (NOT the main wallet) to prove ownership.
+
+**Signing Process**:
+1. Client creates request payload: `[requestId, "auth_verify", params, timestamp]`
+2. Client signs: `signature = sign(keccak256(payload), sessionKeyPrivateKey)`
+3. Client sends request with signature in `sig` array
+
+:::danger Critical Security Requirement
+The `auth_verify` request MUST be signed by the **session key**, not the main wallet. This proves the client controls the session key's private key. Signing with the wrong key will fail authentication.
```

Comment:
It's not true, `auth_verify` request should contain EIP712 signature signed by the main wallet